operations/mimir-mixin/dashboards/writes.libsonnet

```diff
@@ -142,7 +142,22 @@ local filename = 'mimir-writes.json';
       $.row('Ingester')
       .addPanel(
         $.panel('Requests / sec') +
-        $.qpsPanel('cortex_request_duration_seconds_count{%s,route="/cortex.Ingester/Push"}' % $.jobMatcher($._config.job_names.ingester))
+        $.qpsPanel('cortex_request_duration_seconds_count{%s,route="/cortex.Ingester/Push"}' % $.jobMatcher($._config.job_names.ingester)) +
+        if $._config.show_rejected_requests_on_writes_dashboard then {
+          targets: [
+            {
+              legendLink: null,
+              expr: 'sum (rate(cortex_ingester_instance_rejected_requests_total{%s, reason=~"ingester_max_inflight_push_requests|ingester_max_ingestion_rate"}[$__rate_interval]))' % [$.jobMatcher($._config.job_names.ingester)],
```

Comment:
> If `-ingester.limit-inflight-requests-using-grpc-method-limiter` isn't enabled, but `$._config.show_rejected_requests_on_writes_dashboard` is `true`, does this mean we'll count some requests twice on this panel?

Yes. If `-ingester.limit-inflight-requests-using-grpc-method-limiter` is not enabled, they will be counted towards "errors" too.


> If so, is this a likely scenario?

It's new experimental feature in Mimir, I don't expect anyone to have it enabled. That said, we're rolling out `-ingester.limit-inflight-requests-using-grpc-method-limiter` internally, so in our environments this will be the case temporarily. Once we test the new limiting and it proves useful, we would just remove the flags, both CLI flag and config option for dashboards.

> And could we do anything to detect it and not duplicate the requests? (eg. could we add a label to `cortex_ingester_instance_rejected_requests_total` so we can filter to only those requests rejected by the gRPC rate limiter, or emit another per-ingester instance metric that is 1 when the gRPC rate limiter is enabled and use that to multiply `cortex_ingester_instance_rejected_requests_total`?)

We could also use different "reason" when using `-ingester.limit-inflight-requests-using-grpc-method-limiter`, and then add those specific reasons to the dashboard. That solution seemed bit ugly, but perhaps it would be better?

Reply:
>Once we test the new limiting and it proves useful, we would just remove the flags, both CLI flag and config option for dashboards.

I think if this is just a temporary thing while we're testing it, then we don't need to do anything more for this issue - we can just be careful with when / how we enable it.